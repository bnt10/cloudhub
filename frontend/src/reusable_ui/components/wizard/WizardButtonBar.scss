/*
  Wizard Button Bar Styles
  ------------------------------------------------------------------------------
*/

.wizard-button-bar {
  display: inline-flex;
  flex-shrink: 0;
  margin: $ix-marg-b auto $ix-marg-c auto;
  position: relative;
  min-width: 100%;
  justify-content: center;
  align-items: center;
  height: auto;
  flex-wrap: nowrap;
  .btn {
    position: relative;
    white-space: nowrap;
    margin: 0 $ix-marg-a;
  }
}
