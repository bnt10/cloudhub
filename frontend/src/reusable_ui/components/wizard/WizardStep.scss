/*
  Wizard Step Styles
  ------------------------------------------------------------------------------
*/

.wizard-step--container {
  position: relative;
  overflow: hidden;
  margin: $ix-marg-c 0;
  width: 100%;
}

.wizard-step--child {
  padding: $ix-marg-c;
}

.wizard-step--shadow {
  position: absolute;
  min-width: 100%;
  min-height: 100%;
  pointer-events: none;
  z-index: 100;
  box-shadow: inset 0px -11px 8px -10px $g0-obsidian,
    inset 0px 11px 8px -10px $g0-obsidian;
}
