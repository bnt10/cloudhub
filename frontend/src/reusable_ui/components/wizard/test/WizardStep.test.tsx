import React from 'react'
import {shallow} from 'enzyme'

import WizardStep from 'src/reusable_ui/components/wizard/WizardStep'
import WizardButtonBar from 'src/reusable_ui/components/wizard/WizardButtonBar'

describe('WizardStep', () => {
  let wrapper

  const wrapperSetup = (override = {}) => {
    const props = {
      title: 'my wizard step',
      isComplete: () => true,
      isErrored: undefined,
      isSkippableStep: undefined,
      onPrevious: undefined,
      onNext: undefined,
      increment: undefined,
      decrement: undefined,
      tipText: undefined,
      nextLabel: undefined,
      previousLabel: undefined,
      lastStep: undefined,
      ...override,
    }

    const mockChild = 'this is a text'

    return shallow(<WizardStep {...props}>{mockChild}</WizardStep>)
  }

  beforeEach(() => {
    wrapper = wrapperSetup()
  })

  it('mounts without exploding', () => {
    expect(wrapper).toHaveLength(1)
  })

  it('mounts a button bar', () => {
    expect(wrapper.find(WizardButtonBar)).toHaveLength(1)
  })

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('WizardStep handleClickPrevious', () => {
    const newProps = {
      onPrevious: jest.fn(),
      decrement: jest.fn(),
    }

    beforeEach(() => {
      jest.resetAllMocks()
      wrapper = wrapperSetup(newProps)
    })

    it('calls onPrevious on handleClickPrevious', () => {
      const spy = jest.spyOn(newProps, 'onPrevious')

      expect(spy).not.toBeCalled()
      wrapper.instance().handleClickPrevious()
      expect(spy).toBeCalled()
    })

    it('calls decrement on handleClickPrevious', async () => {
      const spy = jest.spyOn(newProps, 'decrement')

      expect(spy).not.toBeCalled()
      await wrapper.instance().handleClickPrevious()
      expect(spy).toBeCalled()
    })
  })
})
