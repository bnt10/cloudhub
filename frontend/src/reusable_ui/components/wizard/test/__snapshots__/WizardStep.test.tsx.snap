// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WizardStep matches snapshot 1`] = `
<Fragment>
  <div
    className="wizard-step--container"
  >
    <div
      className="wizard-step--shadow"
    />
    <FancyScrollbar
      autoHeight={true}
      autoHide={false}
      maxHeight={345.6}
      setScrollTop={[Function]}
      style={Object {}}
    >
      <div
        className="wizard-step--child"
      >
        this is a text
      </div>
    </FancyScrollbar>
  </div>
  <WizardButtonBar
    nextLabel="Next"
    onClickNext={[Function]}
    onClickPrevious={[Function]}
    previousLabel="Previous"
  />
</Fragment>
`;
